src/components/App.jsx

```diff
@@ -0,0 +1,209 @@
+import "../blocks/App.css";
+import { useState, useEffect } from "react";
+import { Route, Routes, Navigate, useNavigate } from "react-router-dom";
+import { SearchContext } from "../contexts/SearchContext";
+import { UserContext } from "../contexts/UserContext";
+import { AppContext } from "../contexts/AppContext";
+import Api from "../utils/api";
+import NewsApi from "../utils/newsApi";
+import { authorize } from "../utils/auth";
+import Nav from "./Nav";
+import Main from "./Main";
+import Footer from "./Footer";
+import ProtectedRoute from "./ProtectedRoute";
+import SavedNews from "./SavedNews";
+import SigninModal from "./SigninModal";
+import SignupModal from "./SignupModal";
+
+export default function App() {
+  const [searchState, setSearchState] = useState({
+    results: [],
+    keyword: "",
+    articlesAvail: 0,
+    articlesShown: 3,
+    loading: false,
+    nothingFound: false,
+  });
+  const [userState, setUserState] = useState({
+    loggedIn: false,
+    email: "",
+    username: "",
+    savedNews: [],
+  });
+  const [activeModal, setActiveModal] = useState("");
+  const [protectedDestination, setProtectedDestination] = useState("");
+  const navigate = useNavigate();
+
+  const api = new Api();
+
+  const newsApi = new NewsApi({
+    baseUrl: "https://nomoreparties.co/news/v2/everything",
+    apiKey: "a16de474931b4e5a83f83ad53ba3df69",
+  });
+
+  useEffect(() => {
+    if (protectedDestination !== "") setActiveModal("signin");
+  }, [protectedDestination]);
+
+  useEffect(() => {
+    document.addEventListener("keydown", (e) => {
+      if (e.key === "Escape") closeActiveModal();
+    });
+
+    document.querySelectorAll(".modal").forEach((modal) => {
+      modal.addEventListener("click", (e) => {
+        if (e.target.classList[0] === "modal") closeActiveModal();
+      });
+    });
+  }, []);
+
+  const handleSearchSubmit = (query) => {
+    setSearchState((currState) => ({
+      ...currState,
+      loading: true,
+      nothingFound: false,
+      articlesShown: 3,
+    }));
+    newsApi
+      .getNewsArticles(query)
+      .then((res) => {
+        setSearchState((currState) => ({
+          ...currState,
+          keyword: query,
+          articlesAvail: res.totalResults,
+          results: res.articles.filter(
+            (article) => article.title !== "[Removed]"
+          ),
+          nothingFound: res.totalResults <= 0 ? true : false,
+        }));
+      })
+      .catch((err) => console.error(err))
+      .finally(() =>
+        setSearchState((currState) => ({ ...currState, loading: false }))
+      );
+  };
+
+  const addSavedArticle = (newArticle) => {
+    api.addArticle(newArticle).then((res) => {
+      const updatedSavedNews = [...userState.savedNews, res];
+      setUserState({
+        ...userState,
+        savedNews: updatedSavedNews,
+      });
+    });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**